File: 品味Java的21種設計模式/dp_cpp/builder/example1/ExportDataModel.cpp
#include "ExportDataModel.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace builder
			{
				namespace example1
				{

					std::string ExportDataModel::getProductId()
					{
						return productId;
					}

					void ExportDataModel::setProductId(std::string productId)
					{
						this->productId = productId;
					}

					double ExportDataModel::getPrice()
					{
						return price;
					}

					void ExportDataModel::setPrice(double price)
					{
						this->price = price;
					}

					double ExportDataModel::getAmount()
					{
						return amount;
					}

					void ExportDataModel::setAmount(double amount)
					{
						this->amount = amount;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/builder/example3/ExportDataModel.java
package cn.javass.dp.builder.example3;
/**
 * 描述输出数据的对象
 */
public class ExportDataModel {
	/**
	 * 产品编号
	 */
	private String productId;
	/**
	 * 销售价格
	 */
	private double price;
	/**
	 * 销售数量
	 */
	private double amount;
	
	public String getProductId() {
		return productId;
	}
	public void setProductId(String productId) {
		this.productId = productId;
	}
	public double getPrice() {
		return price;
	}
	public void setPrice(double price) {
		this.price = price;
	}
	public double getAmount() {
		return amount;
	}
	public void setAmount(double amount) {
		this.amount = amount;
	}
	
}
